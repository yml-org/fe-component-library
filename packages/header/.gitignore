.sass-cache/
*.css.map
*.sass.map
*.scss.map
dist
node_modules
