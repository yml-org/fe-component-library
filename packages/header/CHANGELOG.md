# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.1](https://github.com/yml-org/fe-component-library/compare/@yml-webcl/header@0.1.0...@yml-webcl/header@0.1.1) (2023-03-13)

### Bug Fixes

- **version:** update package versions ([7d8cf7f](https://github.com/yml-org/fe-component-library/commit/7d8cf7f15d4f548555eb74705c83e8125f89486c))

# 0.1.0 (2023-03-13)

### Bug Fixes

- **version:** update package versions ([ebeb6fa](https://github.com/yml-org/fe-component-library/commit/ebeb6faaae123cac74215813d1d1ffaf01fee0cc))
- **version:** update package versions ([d0aeb30](https://github.com/yml-org/fe-component-library/commit/d0aeb30a06f38e6a5b5b2f309f2cd86448a848e8))

### Features

- **storybook:** update documentation for header component ([#23](https://github.com/yml-org/fe-component-library/issues/23)) ([90450d3](https://github.com/yml-org/fe-component-library/commit/90450d3214c9b967eb29210d8fde3d41acb90619))
