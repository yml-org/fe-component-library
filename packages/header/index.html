<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>web components</title>
    <link href="https://vjs.zencdn.net/7.21.1/video-js.css" rel="stylesheet" />
  </head>
  <body>
    <header-component></header-component>
    <script type="module" src="src/header/header.component.ts"></script>
  </body>
</html>
