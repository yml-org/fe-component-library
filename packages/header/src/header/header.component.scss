.btn-tap-color {
  -webkit-tap-highlight-color: transparent;
}