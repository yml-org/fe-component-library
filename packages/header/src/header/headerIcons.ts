import { msg, str} from '@lit/localize';
import { html } from 'lit-html';

export const OpenMenuIcon = html` <svg
  class="block h-6 w-6"
  aria-label=${msg(str`Open menu`)}
  xmlns="http://www.w3.org/2000/svg"
  fill="currentColor"
  viewBox="0 0 24 24"
  stroke-width="1.5"
  stroke="currentColor"
  aria-hidden="true"
>
  <path
    stroke-linecap="round"
    stroke-linejoin="round"
    d="M3.75 6.75h16.5M3.75 12h16.5m-16.5 5.25h16.5"
  />
</svg>`;

export const CloseMenuIcon = html`<svg
  class="block h-6 w-6"
  aria-label=${msg(str`Close menu`)}
  xmlns="http://www.w3.org/2000/svg"
  fill="currentColor"
  viewBox="0 0 24 24"
  stroke-width="1.5"
  stroke="currentColor"
  aria-hidden="true"
>
  <path
    stroke-linecap="round"
    stroke-linejoin="round"
    d="M6 18L18 6M6 6l12 12"
  />
</svg> `;
