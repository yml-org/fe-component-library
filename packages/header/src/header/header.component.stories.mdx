import { html } from 'lit';
import { Story, Meta, Canvas } from '@storybook/addon-docs';
import './header.component';
import {
  withMenuLinks,
  lightModeNavConfig,
  withSlots,
} from '../constants/header.variants';

<Meta title="Header" />

# Header Component

Basic header component without `props`

export const TemplateDefault = () =>
  html`<header-component></header-component>`;

<Canvas columns={1}>
  <Story name="default">{TemplateDefault.bind({})}</Story>
</Canvas>

Header component With `dark` mode and `props`

export const TemplateTheme = (args) => html`<header-component
  .navOptions=${args.navOptions}
>
</header-component>`;

<Canvas columns={1}>
  <Story name="Dark" args={{ navOptions: withMenuLinks }}>
    {TemplateTheme.bind({})}
  </Story>
</Canvas>

Header component With `light` mode and `props`

<Canvas columns={1}>
  <Story name="Light" args={{ navOptions: lightModeNavConfig }}>
    {TemplateTheme.bind({})}
  </Story>
</Canvas>

Header component With `slot` for top right icons and open and close menu icons

export const Template = (args) => html`<header-component
  .navOptions=${args.navOptions}
>
  <div slot="slot-div" style="display:flex;flex-direction:row">
    <svg
      xmlns="http://www.w3.org/2000/svg"
      fill="white"
      viewBox="0 0 24 24"
      stroke-width="1.5"
      stroke="currentColor"
      aria-hidden="true"
      style="width:26px;height:26px;margin-top:3px;margin-right:5px"
    >
      <path
        stroke-linecap="round"
        stroke-linejoin="round"
        d="M14.857 17.082a23.848 23.848 0 005.454-1.31A8.967 8.967 0 0118 9.75v-.7V9A6 6 0 006 9v.75a8.967 8.967 0 01-2.312 6.022c1.733.64 3.56 1.085 5.455 1.31m5.714 0a24.255 24.255 0 01-5.714 0m5.714 0a3 3 0 11-5.714 0"
      />
    </svg>
    <img
      class="h-8 w-8 rounded-full"
      src="https://cdn.icon-icons.com/icons2/1378/PNG/512/avatardefault_92824.png"
      alt=""
    />
  </div>
  <button slot="open">open</button>
  <button slot="close">Close</button>
</header-component>`;

<Canvas columns={1}>
  <Story name="withSlot" args={{ navOptions: withSlots }}>
    {Template.bind({})}
  </Story>
</Canvas>
