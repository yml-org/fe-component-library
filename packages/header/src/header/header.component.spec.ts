import { HeaderComponent } from './header.component';
import { Nav } from '../types/header.component';

const navOptions : Nav = {
  mode: 'dark',
  logo: 'https://ymedialabs.atlassian.net/s/1jmxwi/b/8/d35727372e299c952e88a10ef82bbaf6/_/jira-logo-scaled.png',
  headerText: 'YML',
  openMenuIcon: {
    slotName: '',
  },
  closeMenuIcon: {
    slotName: '',
  },
  menuLinks: [],
  isMenuOpen: false,
  topRightSlot: {
    slotName: 'bell',
  },
};

describe('ymlwebcl-header', () => {
  const HEADER_COMPONENT = 'ymlwebcl-header';
  let headerElement: HeaderComponent;

  const getShadowRoot = (tagName: string): ShadowRoot | null =>
    document.body.getElementsByTagName(tagName)[0].shadowRoot;

  beforeEach(() => {
    headerElement = window.document.createElement(
      HEADER_COMPONENT
    ) as HeaderComponent;
    document.body.appendChild(headerElement);
  });

  afterEach(() => {
    document.body.getElementsByTagName(HEADER_COMPONENT)[0].remove();
  });

  it('matches component snapshot', async () => {
    expect(
      document.body.getElementsByTagName(HEADER_COMPONENT)[0]
    ).toMatchSnapshot();
  });

  it('sets the openMenu to false', async () => {
    headerElement.setAttribute('openMenu', 'false');
    window.document.body.appendChild(headerElement);
    await headerElement.updateComplete;
    const renderedComponent =  document.querySelector('ymlwebcl-header').getAttribute('openMenu');
    expect(renderedComponent).toBe('false');
  });

  it('sets the openMenu to true', async () => {
    headerElement.setAttribute('openMenu', 'true');
    window.document.body.appendChild(headerElement);
    await headerElement.updateComplete;
    const renderedComponent = document.querySelector('ymlwebcl-header').getAttribute('openMenu');
    expect(renderedComponent).toBe('true');
  });

  it('sets the default header hoverTextColor correctly', async () => {
    const headerDivClassList =
      document.body.getElementsByClassName('hover:text-white')[0];
    expect(headerDivClassList).not.toBe(-1);
  });

  it('sets the default header headerTextColor correctly', async () => {
    window.document.body.appendChild(headerElement);
    await headerElement.updateComplete;
    const elemList =
      getShadowRoot(HEADER_COMPONENT).querySelectorAll('.text-white');
    expect(elemList).toHaveLength(2);
  });

  it('sets the  header text correctly', async () => {
    headerElement['navOptions'] = navOptions;
    window.document.body.appendChild(headerElement);
    await headerElement.updateComplete;
    const headingElementText =
      getShadowRoot(HEADER_COMPONENT).querySelector('h2').textContent;
    expect(headingElementText).toBe('YML');
  });

  it('renders the component in mode passed by user', async () => {
    headerElement['navOptions'] = { ...navOptions, mode: 'light' };
    window.document.body.appendChild(headerElement);
    await headerElement.updateComplete;
    const elem = getShadowRoot(HEADER_COMPONENT).querySelector('nav');
    expect(elem.classList.contains('bg-gray-200')).toBe(true);
  });

  it('renders the component passed in slot', async () => {
    headerElement['navOptions'] = navOptions;
    window.document.body.appendChild(headerElement);
    document.querySelector('ymlwebcl-header').innerHTML =
      '<div slot="bell" class="slot-test">Test</div>';
    await headerElement.updateComplete;
    const elem = document
      .querySelector('ymlwebcl-header')
      .querySelector('div');
    expect(elem.classList.contains('slot-test')).toBe(true);
  });

  it('triggers the custom event passed', async () => {
    const func = jest.fn();
    headerElement['navOptions'] = {
      ...navOptions,
      menuLinks: [
        {
          type: 'button',
          label: 'Test',
          eventName: 'my-custom-event',
        },
      ],
    };
    window.document.body.appendChild(headerElement);
    await headerElement.updateComplete;
    document
      .querySelector('ymlwebcl-header')
      .addEventListener('my-custom-event', func);
    const customElem = getShadowRoot(HEADER_COMPONENT);
    const btnElem = customElem.querySelectorAll('button');
    btnElem.forEach((elem) => {
      if (elem.classList.contains('hover:text-white')) {
        elem?.click();
      }
    });
    expect(func.mock.calls).toHaveLength(1);
  });
});
