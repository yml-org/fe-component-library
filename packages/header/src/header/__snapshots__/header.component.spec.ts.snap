// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ymlwebcl-header matches component snapshot 1`] = `
D {
  "_$AL": Map {},
  "_$Do": H {
    "_$AA": <!---->,
    "_$AB": null,
    "_$AH": Tt {
      "_$AD": P {
        "el": <template />,
        "parts": Array [
          Object {
            "ctor": [Function],
            "index": 0,
            "name": "class",
            "strings": Array [
              "",
              "
      ",
              " ",
            ],
            "type": 1,
          },
          Object {
            "index": 5,
            "type": 2,
          },
          Object {
            "index": 8,
            "type": 2,
          },
          Object {
            "ctor": [Function],
            "index": 10,
            "name": "class",
            "strings": Array [
              "flex flex-shrink-0 items-center font-bold  pl-4 ",
              " ",
            ],
            "type": 1,
          },
          Object {
            "index": 12,
            "type": 2,
          },
          Object {
            "index": 15,
            "type": 2,
          },
          Object {
            "ctor": [Function],
            "index": 17,
            "name": "name",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 19,
            "name": "hidden",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "index": 21,
            "type": 2,
          },
        ],
      },
      "_$AM": [Circular],
      "_$AN": undefined,
      "u": Array [
        I {
          "_$AH": Array [
            "bg-black",
            "text-white",
          ],
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <nav
            class="bg-black
      text-white "
            part="nav"
          >
            
        
        
            <div
              class="mx-auto max-w-8xl px-2 sm:px-6 lg:px-4"
            >
              
          
          
              <div
                class="relative flex h-16 items-center justify-between"
              >
                
            
            
                <div
                  class="absolute inset-y-0 left-0 flex items-center sm:hidden"
                  part="menu-icon-container"
                >
                  
              
              
                  <!-- Mobile menu button-->
                  
            
                  <!--?lit$311819298$-->
                  
            
                </div>
                

            
                <div
                  class="flex flex-1 items-center justify-center sm:items-stretch sm:justify-start"
                >
                  
              
                  <div
                    class="flex flex-shrink-0 items-center"
                    part="logo-container"
                  >
                    
                
                    <!--?lit$311819298$-->
                    <img
                      alt="Company logo"
                      class="hidden h-8 w-8 lg:block"
                      part="logo"
                      src="https://ymedialabs.atlassian.net/s/1jmxwi/b/8/d35727372e299c952e88a10ef82bbaf6/_/jira-logo-scaled.png"
                    />
                    
              
                  </div>
                   
         
            
                  <div
                    class="flex flex-1 items-center justify-center sm:items-stretch sm:justify-start"
                  >
                    
              
              
                    <div
                      class="flex flex-shrink-0 items-center font-bold  pl-4 text-white "
                      part="headerText-container"
                    >
                      
                
                      <h2
                        part="headerText"
                      >
                        <!--?lit$311819298$-->
                        YML
                      </h2>
                      
              
                    </div>
                    

              
                    <div
                      class="hidden sm:ml-6 sm:block"
                    >
                      
                
                      <div
                        class="flex space-x-4 pl-10"
                        part="menu-link-container"
                      >
                        
                 
                
                        <!--?lit$311819298$-->
                         
                        <!--?lit$311819298$-->
                        <!--?-->
                        
                
                      </div>
                      
              
                    </div>
                    

            
                  </div>
                  

            
                  <div
                    class="inset-y-0 right-0 flex pr-2 sm:static sm:inset-auto sm:ml-6 sm:pr-0"
                    part="right-icons-container"
                  >
                    
                
                    <slot
                      name="bell"
                    />
                    
            
                  </div>
                  

          
                </div>
                

        
              </div>
              

        
              <!-- Mobile menu, show/hide based on menu state. -->
              

        
              <div
                class="sm:hidden"
                hidden=""
                id="mobile-menu"
              >
                
          
          
                <div
                  class="space-y-1 px-2 pt-2 pb-3"
                  part="menu-link-container-mobile"
                >
                  
          
                  <!--?lit$311819298$-->
                   
                  <!--?lit$311819298$-->
                  <!--?-->
                  
          
                </div>
                

        
              </div>
              

      
            </div>
          </nav>,
          "name": "class",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "strings": Array [
            "",
            "
      ",
            " ",
          ],
          "type": 1,
        },
        H {
          "_$AA": <!--?lit$311819298$-->,
          "_$AB": 
            ,
          "_$AH": Symbol(lit-nothing),
          "_$AM": [Circular],
          "_$AN": undefined,
          "_$Cm": true,
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 2,
        },
        H {
          "_$AA": <!--?lit$311819298$-->,
          "_$AB": 
              ,
          "_$AH": Tt {
            "_$AD": P {
              "el": <template />,
              "parts": Array [
                Object {
                  "ctor": [Function],
                  "index": 0,
                  "name": "src",
                  "strings": Array [
                    "",
                    "",
                  ],
                  "type": 1,
                },
                Object {
                  "ctor": [Function],
                  "index": 0,
                  "name": "alt",
                  "strings": Array [
                    "",
                    "",
                  ],
                  "type": 1,
                },
              ],
            },
            "_$AM": [Circular],
            "_$AN": undefined,
            "u": Array [
              I {
                "_$AH": "https://ymedialabs.atlassian.net/s/1jmxwi/b/8/d35727372e299c952e88a10ef82bbaf6/_/jira-logo-scaled.png",
                "_$AM": [Circular],
                "_$AN": undefined,
                "element": <img
                  alt="Company logo"
                  class="hidden h-8 w-8 lg:block"
                  part="logo"
                  src="https://ymedialabs.atlassian.net/s/1jmxwi/b/8/d35727372e299c952e88a10ef82bbaf6/_/jira-logo-scaled.png"
                />,
                "name": "src",
                "options": Object {
                  "host": [Circular],
                  "isConnected": true,
                  "renderBefore": null,
                },
                "type": 1,
              },
              I {
                "_$AH": "Company logo",
                "_$AM": [Circular],
                "_$AN": undefined,
                "element": <img
                  alt="Company logo"
                  class="hidden h-8 w-8 lg:block"
                  part="logo"
                  src="https://ymedialabs.atlassian.net/s/1jmxwi/b/8/d35727372e299c952e88a10ef82bbaf6/_/jira-logo-scaled.png"
                />,
                "name": "alt",
                "options": Object {
                  "host": [Circular],
                  "isConnected": true,
                  "renderBefore": null,
                },
                "type": 1,
              },
            ],
          },
          "_$AM": [Circular],
          "_$AN": undefined,
          "_$Cm": true,
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 2,
        },
        I {
          "_$AH": Array [
            "text-white",
          ],
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <div
            class="flex flex-shrink-0 items-center font-bold  pl-4 text-white "
            part="headerText-container"
          >
            
                
            <h2
              part="headerText"
            >
              <!--?lit$311819298$-->
              YML
            </h2>
            
              
          </div>,
          "name": "class",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "strings": Array [
            "flex flex-shrink-0 items-center font-bold  pl-4 ",
            " ",
          ],
          "type": 1,
        },
        H {
          "_$AA": <!--?lit$311819298$-->,
          "_$AB": null,
          "_$AH": "YML",
          "_$AM": [Circular],
          "_$AN": undefined,
          "_$Cm": true,
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 2,
        },
        H {
          "_$AA": <!--?lit$311819298$-->,
          "_$AB": 
                ,
          "_$AH": Tt {
            "_$AD": P {
              "el": <template />,
              "parts": Array [
                Object {
                  "index": 0,
                  "type": 2,
                },
              ],
            },
            "_$AM": [Circular],
            "_$AN": undefined,
            "u": Array [
              H {
                "_$AA": <!--?lit$311819298$-->,
                "_$AB": <!--?-->,
                "_$AH": Array [],
                "_$AM": [Circular],
                "_$AN": undefined,
                "_$Cm": true,
                "options": Object {
                  "host": [Circular],
                  "isConnected": true,
                  "renderBefore": null,
                },
                "type": 2,
              },
            ],
          },
          "_$AM": [Circular],
          "_$AN": undefined,
          "_$Cm": true,
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 2,
        },
        I {
          "_$AH": "bell",
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <slot
            name="bell"
          />,
          "name": "name",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
        Nt {
          "_$AH": true,
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <div
            class="sm:hidden"
            hidden=""
            id="mobile-menu"
          >
            
          
          
            <div
              class="space-y-1 px-2 pt-2 pb-3"
              part="menu-link-container-mobile"
            >
              
          
              <!--?lit$311819298$-->
               
              <!--?lit$311819298$-->
              <!--?-->
              
          
            </div>
            

        
          </div>,
          "name": "hidden",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 4,
        },
        H {
          "_$AA": <!--?lit$311819298$-->,
          "_$AB": 
          ,
          "_$AH": Tt {
            "_$AD": P {
              "el": <template />,
              "parts": Array [
                Object {
                  "index": 0,
                  "type": 2,
                },
              ],
            },
            "_$AM": [Circular],
            "_$AN": undefined,
            "u": Array [
              H {
                "_$AA": <!--?lit$311819298$-->,
                "_$AB": <!--?-->,
                "_$AH": Array [],
                "_$AM": [Circular],
                "_$AN": undefined,
                "_$Cm": true,
                "options": Object {
                  "host": [Circular],
                  "isConnected": true,
                  "renderBefore": null,
                },
                "type": 2,
              },
            ],
          },
          "_$AM": [Circular],
          "_$AN": undefined,
          "_$Cm": true,
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 2,
        },
      ],
    },
    "_$AM": undefined,
    "_$AN": undefined,
    "_$Cm": true,
    "options": Object {
      "host": [Circular],
      "isConnected": true,
      "renderBefore": null,
    },
    "type": 2,
  },
  "_$EC": undefined,
  "_$E_": Promise {},
  "_$Ei": undefined,
  "_$El": null,
  "__navOptions": Object {
    "closeMenuIcon": Object {
      "slotName": "",
    },
    "headerText": "YML",
    "isMenuOpen": false,
    "logo": "https://ymedialabs.atlassian.net/s/1jmxwi/b/8/d35727372e299c952e88a10ef82bbaf6/_/jira-logo-scaled.png",
    "menuLinks": Array [],
    "mode": "dark",
    "openMenuIcon": Object {
      "slotName": "",
    },
    "topRightSlot": Object {
      "slotName": "bell",
    },
  },
  "__openMenu": false,
  "enableUpdating": [Function],
  "hasUpdated": true,
  "isUpdatePending": false,
  "renderOptions": Object {
    "host": [Circular],
    "isConnected": true,
    "renderBefore": null,
  },
  "renderRoot": ShadowRoot {
    "_$litPart$": H {
      "_$AA": <!---->,
      "_$AB": null,
      "_$AH": Tt {
        "_$AD": P {
          "el": <template />,
          "parts": Array [
            Object {
              "ctor": [Function],
              "index": 0,
              "name": "class",
              "strings": Array [
                "",
                "
      ",
                " ",
              ],
              "type": 1,
            },
            Object {
              "index": 5,
              "type": 2,
            },
            Object {
              "index": 8,
              "type": 2,
            },
            Object {
              "ctor": [Function],
              "index": 10,
              "name": "class",
              "strings": Array [
                "flex flex-shrink-0 items-center font-bold  pl-4 ",
                " ",
              ],
              "type": 1,
            },
            Object {
              "index": 12,
              "type": 2,
            },
            Object {
              "index": 15,
              "type": 2,
            },
            Object {
              "ctor": [Function],
              "index": 17,
              "name": "name",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 19,
              "name": "hidden",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "index": 21,
              "type": 2,
            },
          ],
        },
        "_$AM": [Circular],
        "_$AN": undefined,
        "u": Array [
          I {
            "_$AH": Array [
              "bg-black",
              "text-white",
            ],
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <nav
              class="bg-black
      text-white "
              part="nav"
            >
              
        
        
              <div
                class="mx-auto max-w-8xl px-2 sm:px-6 lg:px-4"
              >
                
          
          
                <div
                  class="relative flex h-16 items-center justify-between"
                >
                  
            
            
                  <div
                    class="absolute inset-y-0 left-0 flex items-center sm:hidden"
                    part="menu-icon-container"
                  >
                    
              
              
                    <!-- Mobile menu button-->
                    
            
                    <!--?lit$311819298$-->
                    
            
                  </div>
                  

            
                  <div
                    class="flex flex-1 items-center justify-center sm:items-stretch sm:justify-start"
                  >
                    
              
                    <div
                      class="flex flex-shrink-0 items-center"
                      part="logo-container"
                    >
                      
                
                      <!--?lit$311819298$-->
                      <img
                        alt="Company logo"
                        class="hidden h-8 w-8 lg:block"
                        part="logo"
                        src="https://ymedialabs.atlassian.net/s/1jmxwi/b/8/d35727372e299c952e88a10ef82bbaf6/_/jira-logo-scaled.png"
                      />
                      
              
                    </div>
                     
         
            
                    <div
                      class="flex flex-1 items-center justify-center sm:items-stretch sm:justify-start"
                    >
                      
              
              
                      <div
                        class="flex flex-shrink-0 items-center font-bold  pl-4 text-white "
                        part="headerText-container"
                      >
                        
                
                        <h2
                          part="headerText"
                        >
                          <!--?lit$311819298$-->
                          YML
                        </h2>
                        
              
                      </div>
                      

              
                      <div
                        class="hidden sm:ml-6 sm:block"
                      >
                        
                
                        <div
                          class="flex space-x-4 pl-10"
                          part="menu-link-container"
                        >
                          
                 
                
                          <!--?lit$311819298$-->
                           
                          <!--?lit$311819298$-->
                          <!--?-->
                          
                
                        </div>
                        
              
                      </div>
                      

            
                    </div>
                    

            
                    <div
                      class="inset-y-0 right-0 flex pr-2 sm:static sm:inset-auto sm:ml-6 sm:pr-0"
                      part="right-icons-container"
                    >
                      
                
                      <slot
                        name="bell"
                      />
                      
            
                    </div>
                    

          
                  </div>
                  

        
                </div>
                

        
                <!-- Mobile menu, show/hide based on menu state. -->
                

        
                <div
                  class="sm:hidden"
                  hidden=""
                  id="mobile-menu"
                >
                  
          
          
                  <div
                    class="space-y-1 px-2 pt-2 pb-3"
                    part="menu-link-container-mobile"
                  >
                    
          
                    <!--?lit$311819298$-->
                     
                    <!--?lit$311819298$-->
                    <!--?-->
                    
          
                  </div>
                  

        
                </div>
                

      
              </div>
            </nav>,
            "name": "class",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "strings": Array [
              "",
              "
      ",
              " ",
            ],
            "type": 1,
          },
          H {
            "_$AA": <!--?lit$311819298$-->,
            "_$AB": 
            ,
            "_$AH": Symbol(lit-nothing),
            "_$AM": [Circular],
            "_$AN": undefined,
            "_$Cm": true,
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 2,
          },
          H {
            "_$AA": <!--?lit$311819298$-->,
            "_$AB": 
              ,
            "_$AH": Tt {
              "_$AD": P {
                "el": <template />,
                "parts": Array [
                  Object {
                    "ctor": [Function],
                    "index": 0,
                    "name": "src",
                    "strings": Array [
                      "",
                      "",
                    ],
                    "type": 1,
                  },
                  Object {
                    "ctor": [Function],
                    "index": 0,
                    "name": "alt",
                    "strings": Array [
                      "",
                      "",
                    ],
                    "type": 1,
                  },
                ],
              },
              "_$AM": [Circular],
              "_$AN": undefined,
              "u": Array [
                I {
                  "_$AH": "https://ymedialabs.atlassian.net/s/1jmxwi/b/8/d35727372e299c952e88a10ef82bbaf6/_/jira-logo-scaled.png",
                  "_$AM": [Circular],
                  "_$AN": undefined,
                  "element": <img
                    alt="Company logo"
                    class="hidden h-8 w-8 lg:block"
                    part="logo"
                    src="https://ymedialabs.atlassian.net/s/1jmxwi/b/8/d35727372e299c952e88a10ef82bbaf6/_/jira-logo-scaled.png"
                  />,
                  "name": "src",
                  "options": Object {
                    "host": [Circular],
                    "isConnected": true,
                    "renderBefore": null,
                  },
                  "type": 1,
                },
                I {
                  "_$AH": "Company logo",
                  "_$AM": [Circular],
                  "_$AN": undefined,
                  "element": <img
                    alt="Company logo"
                    class="hidden h-8 w-8 lg:block"
                    part="logo"
                    src="https://ymedialabs.atlassian.net/s/1jmxwi/b/8/d35727372e299c952e88a10ef82bbaf6/_/jira-logo-scaled.png"
                  />,
                  "name": "alt",
                  "options": Object {
                    "host": [Circular],
                    "isConnected": true,
                    "renderBefore": null,
                  },
                  "type": 1,
                },
              ],
            },
            "_$AM": [Circular],
            "_$AN": undefined,
            "_$Cm": true,
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 2,
          },
          I {
            "_$AH": Array [
              "text-white",
            ],
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <div
              class="flex flex-shrink-0 items-center font-bold  pl-4 text-white "
              part="headerText-container"
            >
              
                
              <h2
                part="headerText"
              >
                <!--?lit$311819298$-->
                YML
              </h2>
              
              
            </div>,
            "name": "class",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "strings": Array [
              "flex flex-shrink-0 items-center font-bold  pl-4 ",
              " ",
            ],
            "type": 1,
          },
          H {
            "_$AA": <!--?lit$311819298$-->,
            "_$AB": null,
            "_$AH": "YML",
            "_$AM": [Circular],
            "_$AN": undefined,
            "_$Cm": true,
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 2,
          },
          H {
            "_$AA": <!--?lit$311819298$-->,
            "_$AB": 
                ,
            "_$AH": Tt {
              "_$AD": P {
                "el": <template />,
                "parts": Array [
                  Object {
                    "index": 0,
                    "type": 2,
                  },
                ],
              },
              "_$AM": [Circular],
              "_$AN": undefined,
              "u": Array [
                H {
                  "_$AA": <!--?lit$311819298$-->,
                  "_$AB": <!--?-->,
                  "_$AH": Array [],
                  "_$AM": [Circular],
                  "_$AN": undefined,
                  "_$Cm": true,
                  "options": Object {
                    "host": [Circular],
                    "isConnected": true,
                    "renderBefore": null,
                  },
                  "type": 2,
                },
              ],
            },
            "_$AM": [Circular],
            "_$AN": undefined,
            "_$Cm": true,
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 2,
          },
          I {
            "_$AH": "bell",
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <slot
              name="bell"
            />,
            "name": "name",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
          Nt {
            "_$AH": true,
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <div
              class="sm:hidden"
              hidden=""
              id="mobile-menu"
            >
              
          
          
              <div
                class="space-y-1 px-2 pt-2 pb-3"
                part="menu-link-container-mobile"
              >
                
          
                <!--?lit$311819298$-->
                 
                <!--?lit$311819298$-->
                <!--?-->
                
          
              </div>
              

        
            </div>,
            "name": "hidden",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 4,
          },
          H {
            "_$AA": <!--?lit$311819298$-->,
            "_$AB": 
          ,
            "_$AH": Tt {
              "_$AD": P {
                "el": <template />,
                "parts": Array [
                  Object {
                    "index": 0,
                    "type": 2,
                  },
                ],
              },
              "_$AM": [Circular],
              "_$AN": undefined,
              "u": Array [
                H {
                  "_$AA": <!--?lit$311819298$-->,
                  "_$AB": <!--?-->,
                  "_$AH": Array [],
                  "_$AM": [Circular],
                  "_$AN": undefined,
                  "_$Cm": true,
                  "options": Object {
                    "host": [Circular],
                    "isConnected": true,
                    "renderBefore": null,
                  },
                  "type": 2,
                },
              ],
            },
            "_$AM": [Circular],
            "_$AN": undefined,
            "_$Cm": true,
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 2,
          },
        ],
      },
      "_$AM": undefined,
      "_$AN": undefined,
      "_$Cm": true,
      "options": Object {
        "host": [Circular],
        "isConnected": true,
        "renderBefore": null,
      },
      "type": 2,
    },
  },
  "themeOptions": Object {
    "dark": Object {
      "bgColor": "bg-black",
      "bgHoverColor": "hover:bg-gray-300",
      "headerTextColor": "text-white",
      "textColor": "text-gray-300",
      "textHoverColorDesktop": "hover:text-white",
      "textHoverColorMobile": "hover:text-black",
    },
    "light": Object {
      "bgColor": "bg-gray-200",
      "bgHoverColor": "hover:bg-stone-700",
      "headerTextColor": "text-black",
      "textColor": "text-stone-900",
      "textHoverColorDesktop": "hover:text-black",
      "textHoverColorMobile": "hover:text-white",
    },
  },
}
`;
